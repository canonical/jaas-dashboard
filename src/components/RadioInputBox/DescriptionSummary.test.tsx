import { render, screen } from "@testing-library/react";

import DescriptionSummary from "./DescriptionSummary";

describe("DescriptionSummary", () => {
  it("returns a details component if the description is longer than 30 characters", () => {
    const description =
      "The number of coding chunks, i.e. the number of additional chunks computed by the encoding functions. If there are 2 coding chunks, it means 2 OSDs can be out without losing data. ";
    const { container } = render(
      <DescriptionSummary description={description} />
    );
    expect(container).toMatchSnapshot();
  });

  it("returns just the text if the description is shorter than 30 characters", () => {
    const description = "The name of the profile";
    render(<DescriptionSummary description={description} />);
    expect(screen.getByText(description)).toBeInTheDocument();
    expect(
      document.querySelector(".radio-input-box__details")
    ).not.toBeInTheDocument();
  });

  it("returns null if no description is provided", () => {
    render(<DescriptionSummary description={undefined} />);
    expect(
      document.querySelector(".radio-input-box__details")
    ).not.toBeInTheDocument();
  });
});
