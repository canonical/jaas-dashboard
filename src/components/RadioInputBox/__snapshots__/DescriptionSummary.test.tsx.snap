// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DescriptionSummary returns a details component if the description is longer than 30 characters 1`] = `
<div>
  <details
    class="radio-input-box__details"
  >
    <summary
      class="radio-input-box__summary"
    >
      <span
        class="radio-input-box__summary-description"
      >
        The number of coding chunks, i.e. the number of additional chunks computed by the encoding functions. If there are 2 coding chunks, it means 2 OSDs can be out without losing data. 
      </span>
      Â 
    </summary>
    <span
      class="radio-input-box__details-description"
    >
      The number of coding chunks, i.e. the number of additional chunks computed by the encoding functions. If there are 2 coding chunks, it means 2 OSDs can be out without losing data. 
    </span>
  </details>
</div>
`;
